### PR TITLE
Installing Pelican/OSDF Origin via RPM (SOFTWARE-5962)

### PR DESCRIPTION
**Pull request by matyasselmeci**

### CODE REVIEW

File: docs/data/osdf/install-origin-rpm.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    In OSG 23, the Pelican-based OSDF RPMs are only available in the "osg-upcoming" repositories.
+
+!!! note "Host certificates"
+    Origins should use a CA that is accepted by major browsers and operating systems,
+    such as InCommon RSA or [Let's Encrypt](../../security/host-certs/lets-encrypt).
+    IGTF certs are not recommended because clients are not configured to accept them by default.
+    
+    The following locations should be used (note that they are in separate directories):
+    
+    * **Host Certificate**: `/etc/pki/tls/certs/pelican.crt`
```
Comment (matyasselmeci) [inline docs/data/osdf/install-origin-rpm.md:53]: You need a certificate _chain_, not just a certificate, so that needs to be added.

**Comment by matyasselmeci** [comment]
Things that are missing:

- Some of the configuration is still up in the air as we learn more from
  supporting JLab/GLBRC/others, and as we revamp the config we ship

- Registration instructions could use improvement

- No validation instructions yet

File: docs/data/osdf/install-origin-rpm.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+Before starting the installation process, consider the following requirements:
+
+* __Operating system:__ A RHEL 8 or RHEL 9 or compatible operating systems.
+* __User IDs:__ If they do not exist already, the installation will create the Linux user ID `xrootd` for running daemons.
+* __Host certificate:__ Required for authentication.  See note below.
+* __Network ports:__ The origin service requires the following ports open:
+  * Inbound TCP port 8443 for file access via the HTTP(S) and XRoot protocols.
+  * (Optional) Inbound TCP port 8444 for access to the web interface for monitoring and configuration;
+    if enabled, this should be restricted to the LAN or management network.
```
Comment (brianhlin) [inline docs/data/osdf/install-origin-rpm.md:32]:
```suggestion
    if enabled, consider restricting access from your LAN.
```

Management network implies BMC / IPMI

File: docs/data/osdf/install-origin-rpm.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+You must configure the following:
+```
+XRootD:
+  Sitename: <RESOURCE NAME REGISTERED WITH OSG>
+```
+
+In addition, you must tell Pelican the data to export to the federation.
+An origin may export one or more directory trees, or one or more S3 buckets -- follow one of the sections below.
+(An single origin cannot export both a bucket and a directory tree.)
```
Comment (brianhlin) [inline docs/data/osdf/install-origin-rpm.md:83]:
```suggestion
A single origin cannot export both a bucket and a directory tree.
```

File: docs/data/osdf/install-origin-rpm.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+      S3AccessKeyfile:  "<PATH TO S3 BUCKET ACCESS KEY>"
+      S3SecretKeyfile:  "<PATH TO S3 BUCKET SECRET KEY>"
+      Capabilities:     # Add or remove as desired
+        - Reads         # Enable authenticated reading of objects from the bucket through a cache
+        - PublicReads   # Enable unauthenticated reading of objects from the bucket through a cache
+        - DirectReads   # Enable reading objects from the bucket
+                        # without going through a cache
+        - Listings      # Enable listing bucket items
+        - Writes        # Enable writing to objects in the bucket
+```
```
Comment (brianhlin) [inline docs/data/osdf/install-origin-rpm.md:130]: I think it makes the most sense to replace these sections with links to

- https://docs.pelicanplatform.org/federating-your-data/s3-backend
- And maybe https://docs.pelicanplatform.org/federating-your-data#origin-and-namespace-capabilities? Keeping the POSIX example around probably makes sense, actually

**Review by brianhlin — COMMENTED** [review]
Seeing as how Mat's leaving for vacation shortly, this review is mostly intended as notes for me for what I think needs to be done with this doc. The main things are:

1. A separate section near the top for the procedure to upgrade from a proto-Pelican origin
2. Thinking about the least painful way to present registration since users will have to do it in two spots (registry, topology)
3. Fleshing out the configuration section: add section for multiuser, add section about logging

**Comment by matyasselmeci** [comment]
I have a second take incorporating your comments and also things from the cache installation via RPM docs, ready for review.

One thing that's missing from the doc is picking a namespace. Since that's going to be common with the 'installation via container' docs, I think that should go into a separate doc (some Pelican/OSDF equivalent of the [Getting VO Data into the OSDF doc](https://osg-htc.org/docs/data/stashcache/vo-data/))

File: docs/data/osdf/install-origin-rpm.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
+title: Installing the OSDF Origin by RPM
+
+Installing the OSDF Origin
+==========================
+
+This document describes how to install a Pelican-based Open Science Data Federation (OSDF) Origin service via RPMs.
+This service allows an organization to export its data to the Data Federation.
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply docs/data/osdf/install-origin-rpm.md]: You fixed this in your branch

File: docs/data/osdf/install-origin-rpm.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+!!! note
+    The origin must be registered with the OSG prior to joining the data federation.
+    You may start the registration process prior to finishing the installation by [using this link](#registering-the-origin) 
+    along with information like:
+
+    * Resource name and hostname
+    * VO associated with this origin server (which will be used to determine the origin's namespace prefix)
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply docs/data/osdf/install-origin-rpm.md]: You fixed this in your branch

File: docs/data/osdf/install-origin-rpm.md (hunk 2 above)
Comment (matyasselmeci) [reply docs/data/osdf/install-origin-rpm.md]: Copied the language from the cache doc, which doesn't have 'management network'.

File: docs/data/osdf/install-origin-rpm.md (hunk 4 above)
Comment (matyasselmeci) [reply docs/data/osdf/install-origin-rpm.md]: You fixed this in your branch

File: docs/data/osdf/install-origin-rpm.md (hunk 3 above)
Comment (matyasselmeci) [reply docs/data/osdf/install-origin-rpm.md]: You fixed this in your branch

File: docs/data/osdf/install-origin-rpm.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+  Exports:
+    # You may have one or more of the following block:
+    - FederationPrefix: "<EXTERNAL OSDF NAMESPACE>"
+      StoragePrefix:    "<LOCAL FILESYSTEM DIRECTORY>"
+      Capabilities:    # Add or remove as desired
+        - Reads        # Enable authenticated reading of objects from under the directory tree through a cache
+        - PublicReads  # Enable unauthenticated reading of objects from under the directory tree through a cache
+        - DirectReads  # Enable reading objects from under the directory tree without going through a cache
+        - Listings     # Enable directory listings
+        - Writes       # Enable writing to files in the directory tree
```
Comment (brianhlin) [inline docs/data/osdf/install-origin-rpm.md:98]: As we discussed, we should turn this into example config in the packaging. I think I remembered putting this together https://github.com/opensciencegrid/tiger-osg-config/blob/master/manifests/base/pelican-origin-yaml-config/50-instance.yaml

**Review by brianhlin — APPROVED** [review]
LGTM

File: docs/data/osdf/install-origin-rpm.md (hunk 7 above)
Comment (brianhlin) [reply docs/data/osdf/install-origin-rpm.md:98]: Actually, we really should just use what's in upstream. I forget why we didn't just do that

- https://github.com/PelicanPlatform/pelican/blob/main/systemd/examples/20-origin-exports.yaml
- https://github.com/PelicanPlatform/pelican/blob/main/systemd/examples/60-origin-multiuser.yaml

Comment (matyasselmeci) [reply docs/data/osdf/install-origin-rpm.md:98]: As we discussed, I'll take care of that in a future PR.

**Comment by matyasselmeci** [comment]
Made public in #222

**Pull request merged**